lib/ash/actions/read/read.ex

```diff
@@ -1972,6 +1972,11 @@ defmodule Ash.Actions.Read do
            (page[:count] != false and action.pagination.countable == :by_default))
 
     cond do
+      # Here it returns nil.
+      # Is it making an assumption that is not satisfied later?
+      # Commenting this and interrogating the datalayer below would result in
+      # the Ets datalayer failing since `run_aggregate_query_with_lateral_join`
+      # is not supported, and Postgres failing when compiling Ecto subqueries.
```

Comment:
For relationship count it's normal it returns `nil` here, it should add an aggregate from the parent [here](https://github.com/ash-project/ash/blob/b3a3d3d406b71a9f5a0f8e183f144d06f481fb37/lib/ash/actions/read/read.ex#L486). I'm guessing that since the read for the parent happens in the `update` action, it doesn't pass through here?

Reply:
Thanks, you pointed me in the right direction: I managed to make the test pass.

I'll double check my solution and add a couple more tests before pushing a new commit.

Reply:
The fix seems actually straightforward, but I took the chance add a bunch of tests on the `load` statements for actions (create, update, destroy).